deepchem/models/torch_models/chemberta.py

```diff
@@ -0,0 +1,141 @@
+    >>> tokenizer_path = "seyonec/PubChem10M_SMILES_BPE_60k"
+    >>> pretrain_model = Chemberta(task='pretraining', model_dir=pretrain_model_dir, tokenizer_path=tokenizer_path, mode='mlm')
+    >>> pretraining_loss = pretrain_model.fit(dataset, nb_epoch=1)
+
+    >>> # finetuning in regression mode
+    >>> finetune_model_dir = os.path.join(tempdir, 'finetune-model')
+    >>> finetune_model = Chemberta(task='finetuning', model_dir=finetune_model_dir, tokenizer_path=tokenizer_path, mode='regression')
+    >>> finetune_model.load_from_pretrained(pretrain_model_dir)
+    >>> finetuning_loss = finetune_model.fit(dataset, nb_epoch=1)
+
+    >>> # prediction and evaluation
+    >>> result = finetune_model.predict(dataset)
+    >>> eval_results = finetune_model.evaluate(dataset, metrics=dc.metrics.Metric(dc.metrics.mae_score))
+
+
+    Reference
+    ---------
+    .. Ahmad, Walid, et al. "Chemberta-2: Towards chemical foundation models." arXiv preprint arXiv:2209.01712 (2022).
+    """
+
+    def __init__(self,
+                 task: str,
+                 mode: Optional[str] = None,
+                 tokenizer_path: str = 'seyonec/PubChem10M_SMILES_BPE_60k',
+                 n_tasks: int = 1,
+                 **kwargs):
+        self.mode = mode
+        self.task = task
+        self.n_tasks = n_tasks
+        tokenizer = RobertaTokenizerFast.from_pretrained(tokenizer_path)
+        if task == 'pretraining':
+            if mode == 'mlm':
+                config = RobertaConfig(vocab_size=tokenizer.vocab_size)
+                model = RobertaForMaskedLM(config)
+            elif mode == 'mtr':
+                config = RobertaConfig(vocab_size=tokenizer.vocab_size,
+                                       problem_type='regression',
+                                       num_labels=n_tasks)
+                model = RobertaForSequenceClassification(config)
+        elif task == 'finetuning':
+            assert mode, 'specify finetuning mode - classification or regression'
+            if mode == 'regression':
+                config = RobertaConfig(vocab_size=tokenizer.vocab_size,
+                                       problem_type='regression',
+                                       num_labels=n_tasks)
+                model = RobertaForSequenceClassification(config)
+            elif mode == 'classification':
+                config = RobertaConfig(vocab_size=tokenizer.vocab_size)
+                model = RobertaForSequenceClassification(config)
+            else:
+                raise ValueError('invalid mode specification')
+        else:
+            raise ValueError('invalid task specification')
+
+        super(Chemberta, self).__init__(model=model,
+                                        task=task,
+                                        tokenizer=tokenizer,
+                                        **kwargs)
+
+    def _prepare_batch(self, batch: Tuple[Any, Any, Any]):
```

Comment:
Ok to move this to HFModel in a separate PR, but we should eventually create a BatchPreparer API to make this logic part of the public API

Reply:
Moved to separate PR #3389